Sources/Sentry/SentryScope.m

```diff
@@ -52,7 +52,7 @@
 /**
  * Set the fingerprint of an event to determine the grouping
  */
-@property (atomic, strong) NSArray<NSString *> *_Nullable fingerprintArray;
+@property (nonatomic, strong) NSMutableArray<NSString *> *fingerprintArray;
```

Comment:
properties are atomic by default, why is this (and the other properties) `nonatomic` explicitly?

Internet says:

> For nonatomic properties, the underlying pointer of a read value could be released when a new value is being written at the same time.

Reply:
I read once or twice that `atomic` on iOS can become a performance problem:

> If you’ve been developing for iOS at all, you’ll notice that all properties are declared nonatomic. The reason is that, historically, the locking introduces such an overhead on iOS that it becomes a performance problem. Usually, atomicity is not required anyway, since it does not ensure thread safety, which usually requires a deeper level of locking.
 
Galloway, Matt. Effective Objective-C 2.0 (Effective Software Development Series) (p. 32). Pearson Education. Kindle Edition. 

I thought that when we have our own locking strategy in place, `atomic` could slow down our code. After running some benchmarks now, I don't see a difference on iOS or macOS. As we make the code thread-safe I changed all arrays to `atomic` now.